plugins/modules/foreman_hostgroup.py

```diff
@@ -312,6 +317,12 @@ def main():
             updated_name=dict(),
         ),
     )
+
+    puppetclass_spec = dict(
+        hostgroup=dict(type='entity', flat_name='hostgroup_id'),
```

Comment:
I think, this should not be in the entity_spec, but be provided as `scope` dictionary to the the `params` variable of `find*` and `ensure*`